app/models/lookup_value.rb

```diff
@@ -53,13 +55,17 @@ def validate_and_cast_value
   end
 
   def validate_regexp
-    return true unless (lookup_key.validator_type == 'regexp')
-    errors.add(:value, _("is invalid")) and return false unless (value =~ /#{lookup_key.validator_rule}/)
+    return true if (lookup_key.validator_type != 'regexp' || (lookup_key.contains_erb?(value) && Setting[:interpolate_erb_in_parameters]))
```

Comment:
You can add this check in the validation directly and avoid getting in the method right? Something like:

```ruby
if Setting[:interpolate_erb_in_parameters]
  validate :validate_list, :unless => -> { |lookup_value| lookup_value.lookup_key.validator_type != 'list' },
                           :if     => -> { |lookup_value| lookup_value.lookup_key.contains_erb?(lookup_value.value) }
  validate :validate_regexp, :unless => -> { |lookup_value| lookup_value.lookup_key.validator_type != 'regexp' },
                             :if     => -> { |lookup_value| lookup_value.lookup_key.contains_erb?(lookup_value.value) }
end
```

If possible, can you consider taking out the validation and before_validation definitions, `validate_regexp` `validate_list` and `validate_and_cast_value` into their own shared concern, and include the concern in LookupKey and LookupValue? 

This is relatively complex logic that is duplicated, and I'm afraid future changes into the code might bring bugs due to changes being in one place and not the other, it makes it difficult to maintain (people have to be on the look out for duplicated code and change it everywhere).